src/conn/conn_ckpt.c

```diff
@@ -235,7 +235,7 @@ __wt_checkpoint_signal(WT_SESSION_IMPL *session, wt_off_t logsize)
 
     conn = S2C(session);
     WT_ASSERT(session, WT_CKPT_LOGSIZE(conn));
-    if (logsize >= conn->ckpt_logsize && !conn->ckpt_signalled) {
+    if (logsize >= conn->ckpt_logsize && !conn->ckpt_signalled && conn->ckpt_cond != NULL) {
```

Comment:
If the capacity server is being destroyed at the correct time during shutdown this null check is no longer required.